### ISSUE TITLE
Feature Services

### ISSUE THREAD

**Issue by stdavis** [agrc/wfrc-bike-map#5]
We need to figure out where the feature services that back this app will be hosted. The data is contained in both the [Roads](https://opendata.gis.utah.gov/datasets/utah::utah-roads/about) and [Trails and Pathways](https://opendata.gis.utah.gov/datasets/utah::utah-trails-and-pathways/about) SGID datasets.

Action items:
- [x] create existing feature service in UGRC AGOL org
- [x] create planned feature service in UGRC AGOL org
- [x] repoint the [staging web map](https://utah.maps.arcgis.com/apps/mapviewer/index.html?webmap=6ad72cac07b6464f9542e7daaa884b44) to the newly created services (a production web map will be created as part of the [pre-production checklist](https://github.com/orgs/agrc/projects/21/views/1?pane=issue&itemId=100180612))
- [x] update [josh's code](https://github.com/WFRCAnalytics/Bike-App-Create-Features) to overwrite the newly created feature services above
- [x] hand off code to @eneemann so he can plug it into the monthly roads update process

## Editing

Possible branch versioning via the UGRC portal instance.

**Comment by stdavis** [comment]
@bert_granberg is going to get us an ETL script and then we will push out the feature services.

**Comment by bert-granberg** [comment]
> @bert_granberg is going to get us an ETL script and then we will push out the feature services.

@joshua-reynolds is planning to do the ETL script that brings the road centerlines features with Bike_* attributes together with the  trails/pathways features. Basically deriving one new feature class with the needed data elements from from the two feature classes. The end product would essentially be "bikeways" (on street + pathways bike routes in the real world). The on street would need the type of bike facility and the level of traffic stress.

**Comment by joshua-reynolds** [comment]
@bert-granberg Don't we already have this in the bike layer of the [multimodal network](https://gis.utah.gov/products/sgid/transportation/multimodal-network/)?

**Comment by nathankota** [comment]
- One service for existing infrastructure
- One service for planned

**Comment by stdavis** [comment]
A preliminary feature service from @joshua-reynolds this morning:

https://wfrc.maps.arcgis.com/home/item.html?id=cc0db244166f486891c02b6d306165fd

This schema is as follows:
'NAME' - name of the street or trail
'CARTOCODE' - cartographic code, 
'SIDE_1' - direction (N,S,W, E), 
'SIDE_2' -  direction (N,S,W, E), 
'TYPE' - bike facility type , 
'STATUS' - existing or planned, 
'SOURCE' - roads or trails, 
'SOURCE_ID' - id for joining back to global id of source data

*highlighted fields have unofficial data

**Comment by joshua-reynolds** [comment]
The script is complete and can be found here:
https://github.com/WFRCAnalytics/Bike-App-Create-Features

**Comment by stdavis** [comment]
The latest draft web map is here: https://wfrc.maps.arcgis.com/apps/mapviewer/index.html?webmap=1da30e08d4ce4950bb44662e5c9a54fb

**Comment by joshua-reynolds** [comment]
New bike feature update:
- added a dissolve to reduce the amount of features
- cities and counties attribute has been fixed
- added level of stress

**Comment by stdavis** [comment]
We decided that UGRC would host the data to back this app in SGID on ArcGIS (not internal or open sgid) as well as wiring up the script via palletjack.

**Comment by stdavis** [comment]
Action items:
- [ ] create feature services in UGRC AGOL org for hosting dev and prod versions of existing and planned versions of this data
- [ ] port [josh's code](https://github.com/WFRCAnalytics/Bike-App-Create-Features) to a skid within this project and wire up to the newly created AGOL items
- [ ] repoint the [staging web map](https://utah.maps.arcgis.com/apps/mapviewer/index.html?webmap=6ad72cac07b6464f9542e7daaa884b44) to the newly created services (a production web map will be created as part of the [pre-production checklist](https://github.com/orgs/agrc/projects/21/views/1?pane=issue&itemId=100180612))

**Comment by stdavis** [comment]
@eneemann Do you have any preferences where I put these feature services (existing & planned bike features) in AGOL? I was thinking that there should be a "Bikeways" folder that would contain the two feature services and the two web maps that will support the app.

**Comment by eneemann** [comment]
Yeah, I think a "Bikeways" folder makes sense to keep everything together.

**Comment by stdavis** [comment]
Helpful docs on how to implement the linear directional mean stuff since we can't use arcpy in a skid: https://pro.arcgis.com/en/pro-app/latest/tool-reference/spatial-statistics/h-how-linear-directional-mean-spatial-statistics-w.htm. It looks narly to me. I may need @jacobdadams help.

Here's the AADT data that is used as an input: https://opendata.gis.utah.gov/datasets/uplan::aadt-rounded/about

**Comment by stdavis** [comment]
I spoke with @eneemann this morning and we decided that it makes more sense to integrate Josh's code into the monthly roads update process when they build other derivative products (statewide road network, multimodal network, etc.) rather than porting it to a skid.

**Comment by stdavis** [comment]
Production Services:
[Utah Bikeways](https://utah.maps.arcgis.com/home/item.html?id=ef79d0e5b12a414996dfcd461921c952)
[Utah Planned Bikeways](https://utah.maps.arcgis.com/home/item.html?id=f974911ed47a4d36a4999be0990fd127)

**Comment by eneemann** [comment]
I ran the script and it appears to have successfully updated the production services.  We will build this into our road update process for regular (monthly) updates.
